src/plugins/data_source/server/plugin.ts

```diff
@@ -84,17 +120,21 @@ export class DataSourcePlugin implements Plugin<DataSourcePluginSetup, DataSourc
   private createDataSourceRouteHandlerContext = (
     dataSourceService: DataSourceServiceSetup,
     cryptographyClient: CryptographyClient,
-    logger: Logger
+    logger: Logger,
+    auditorFactory: AuditorFactory
   ): IContextProvider<RequestHandler<unknown, unknown, unknown>, 'dataSource'> => {
     return (context, req) => {
       return {
         opensearch: {
           getClient: (dataSourceId: string) => {
             try {
+              const auditor = auditorFactory.asScoped(req);
+
```

Comment:
we should use auditTrail service's `asScoped()` function to get auditor, auditTrail service provides function to only create 1 auditor for a given request even if we have multiple events for a request